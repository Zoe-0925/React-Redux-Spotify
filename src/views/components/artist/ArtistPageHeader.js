import React from 'react';
import PropTypes from 'prop-types';

export default function ArtistPageHeader(props) {
    return (
        <div className="ArtistPageHeader">
            <div className="playlistPageImage">
                <img
                    src={props.imgSrc}
                    alt="artist"
                />
            </div>
            <div className="playlistPageContent">
                <p className="smallText uppercase bold">VERIFIED ARTIST</p>
                <h1>{props.artistName}</h1>
                {props.saved &&
                <div className="follow-button unfilledIcon" onClick={props.toggleFollow}>Follow</div>}
                {!props.saved &&
                <div className="follow-button coloredIcon" onClick={props.toggleFollow}>UnFollow</div>}
            </div>
        </div>
    )
}

ArtistPageHeader.prototype={
    imgSrc:PropTypes.string,
    artistName:PropTypes.string,
}