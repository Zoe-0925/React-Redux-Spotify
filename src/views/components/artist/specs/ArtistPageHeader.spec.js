import React from 'react';
import renderer from 'react-test-renderer';
import ArtistPageHeader from "../ArtistPageHeader"

const imgSrc = "https://i.scdn.co/image/ab67616d0000b2736040effba89b9b00a6f6743a"
const artistName="Lady Gaga"


it('renders correctly', () => {
    const component = renderer.create(<ArtistPageHeader imgSrc={imgSrc} artistName={artistName}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});