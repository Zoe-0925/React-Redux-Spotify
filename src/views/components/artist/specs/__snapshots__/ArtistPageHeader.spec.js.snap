// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ArtistPageHeader"
>
  <div
    className="playlistPageImage"
  >
    <img
      alt="artist"
      src="https://i.scdn.co/image/ab67616d0000b2736040effba89b9b00a6f6743a"
    />
  </div>
  <div
    className="playlistPageContent"
  >
    <p
      className="smallText uppercase bold"
    >
      VERIFIED ARTIST
    </p>
    <h1>
      Lady Gaga
    </h1>
  </div>
</div>
`;
